### ISSUE TITLE
BUG: random with array-like parameters

### ISSUE THREAD

**Issue by to266** [dask/dask#1326]
the wrapping of `np.random.RandomState` uses the original arguments to figure out the dtype changing the `size` keyword. `numpy` actually requires the `size` and (if broadcastable) init arguments to be of the same size, so it fails:

```python
In [1]: import dask.array as da

In [2]: arr = da.full((3,2), 3, chunks=2)
/home/to266/anaconda3/lib/python3.5/site-packages/numpy/core/numeric.py:294: FutureWarning: in the future, full((3, 2),
 3) will return an array of dtype('int64')
  format(shape, fill_value, array(fill_value).dtype), FutureWarning)

In [3]: da.random.poisson(arr, chunks=arr.chunks, size=arr.size)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-fae6fadd81f5> in <module>()
----> 1 da.random.poisson(arr, chunks=arr.chunks, size=arr.size)

/home/to266/anaconda3/lib/python3.5/site-packages/dask/array/random.py in poisson(self, lam, size, chunks)
    186     def poisson(self, lam=1.0, size=None, chunks=None):
    187         return self._wrap(np.random.RandomState.poisson, lam,
--> 188                          size=size, chunks=chunks)
    189
    190     @doc_wraps(np.random.RandomState.power)

/home/to266/anaconda3/lib/python3.5/site-packages/dask/array/random.py in _wrap(self, func, *args, **kwargs)
     61         kw = kwargs.copy()
     62         kw['size'] = (0,)
---> 63         dtype = func(np.random.RandomState(), *args, **kw).dtype
     64
     65         # Build graph

mtrand.pyx in mtrand.RandomState.poisson (numpy/random/mtrand/mtrand.c:25744)()

mtrand.pyx in mtrand.discd_array (numpy/random/mtrand/mtrand.c:7518)()

ValueError: shape mismatch: objects cannot be broadcast to a single shape
```

**Comment by jcrist** [comment]
The random functions in dask array aren't designed to handle non-scalar inputs. This could be done though, but it will mildly complicate things. A few options:

- Check for non-scalar parameters and throw an informative error (easy)
- Implement the broadcasting in dask array, so that the above works (a bit harder)

Is this a feature that you need? Note that your specific example above is equivalent to
`da.random.poisson(3, size=(3, 2), chunks=2)`, which does work.

**Comment by to266** [comment]
It is a feature I need, and the example was just a toy thing that raised the error.

Just wanted to bring the bug to your attention. If no one fixes this, I suppose I'll just attempt to PR the broadcasting, hopefully somewhat soon...

**Comment by jcrist** [comment]
Fixed by #1327. Closing.
